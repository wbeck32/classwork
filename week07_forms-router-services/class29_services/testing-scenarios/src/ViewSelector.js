import React, { Component } from 'react';
import { RadioGroup, RadioButton } from 'react-radio-buttons';

// these would be imported
function Thumbnail() { return <div>Thumbnail</div>; }
function Gallery() { return <div>Gallery</div>; }
function Detail() { return <div>Detail</div>; }

const Views = {
    thumbnail: Thumbnail,
    gallery: Gallery,
    detail: Detail,
}

const views = Object.keys(Views);

export default class ViewSelector extends Component {
    constructor(props) {
        super(props);
        this.state = { 
          view: views[0]
        };
    }

    handleChange = view => {
      this.setState({ view });
    } 

    render() {
      const { view } = this.state;
      return (
        <div>
          <RadioButtons views={views}/>
          <ViewDisplay view={view} images={this.props.images}/>
        </div>
      );
    }
}

function RadioButtons({ views, onChange }) {
  return (
    <RadioGroup onChange={onChange} horizontal>  
      {views.map(view => (
        <RadioButton key={view} value={view}>
          {view}
        </RadioButton>
      ))}
    </RadioGroup>
  );
}

export function ViewDisplay({ view, images }) {
    const View = Views[view];
    return <View images={images}/>
}