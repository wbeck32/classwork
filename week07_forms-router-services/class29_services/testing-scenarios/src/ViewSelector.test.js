import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ViewSelector, { ViewDisplay } from './ViewSelector';

describe('ViewDisplay', () => {

  function testView(view) {
    it(`shows ${view}`, () => {
      const wrapper = shallow(<ViewDisplay view={view} images={[]} />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  }

  testView('gallery');
  testView('detail');
  testView('thumbnail');
});

describe('ViewSelector', () => {

  it('toggles to correct view', () => {
    const wrapper = shallow(<ViewSelector images={[]} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
    
    const component = wrapper.instance();
    
    component.handleChange('gallery');
    wrapper.update();
    expect(toJSON(wrapper)).toMatchSnapshot();
    
    component.handleChange('detail');
    wrapper.update();
    expect(toJSON(wrapper)).toMatchSnapshot();

  });
})
