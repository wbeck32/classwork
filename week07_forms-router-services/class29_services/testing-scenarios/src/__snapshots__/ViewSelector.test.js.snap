// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewDisplay shows detail 1`] = `
<Detail
  images={Array []}
/>
`;

exports[`ViewDisplay shows gallery 1`] = `
<Gallery
  images={Array []}
/>
`;

exports[`ViewDisplay shows thumbnail 1`] = `
<Thumbnail
  images={Array []}
/>
`;

exports[`ViewSelector toggles to correct view 1`] = `
<div>
  <RadioButtons
    views={
      Array [
        "thumbnail",
        "gallery",
        "detail",
      ]
    }
  />
  <ViewDisplay
    images={Array []}
    view="thumbnail"
  />
</div>
`;

exports[`ViewSelector toggles to correct view 2`] = `
<div>
  <RadioButtons
    views={
      Array [
        "thumbnail",
        "gallery",
        "detail",
      ]
    }
  />
  <ViewDisplay
    images={Array []}
    view="gallery"
  />
</div>
`;

exports[`ViewSelector toggles to correct view 3`] = `
<div>
  <RadioButtons
    views={
      Array [
        "thumbnail",
        "gallery",
        "detail",
      ]
    }
  />
  <ViewDisplay
    images={Array []}
    view="detail"
  />
</div>
`;
